subseas_flex_fcst/maproom.py

```diff
@@ -45,26 +48,75 @@
+
+        dateFormat = re.compile(".*-.*-.*")
+        startDateStr = list(filter(dateFormat.match,fileCompList))
+        startDate = startDateStr[0]
+        #startDate = datetime.strptime(startDateStr[0],
+        #'%b-%d-%Y').date()
+        if leadTimeWk[0] == 'wk1':
+            leadTimeDate = 'Week 1'#'Apr 2' #should it be made into dates or kept as wk number?
+        elif leadTimeWk[0] == 'wk2':
+            leadTimeDate = 'Week 2'#'Apr 9'
+        elif leadTimeWk[0] == 'wk3':
+            leadTimeDate = 'Week 3'#'Apr 16'
+        elif leadTimeWk[0] == 'wk4':
+            leadTimeDate = 'Week 4'#'Apr 23'
+        leadTimeDict = {'L':[leadTimeDate]}
+
+        ds = cptio.open_cptdataset(filesNameList[idx])
+        if dof == True:
+            dofVar = float(ds["tp_pred_err_var"].attrs["dof"])
+            ds = ds.assign(dof=dofVar)
+            #dofDict = {'dof':[dof]}
+            #ds =ds.assign_coords(dof=("dof",[dof]))
+            #ds = ds.expand_dims(dofDict)
+        if len(ds['T']) == 1:
+            startDate
+            ds['T'] = [startDate] #for obs data there are many dates
+        ds = ds.expand_dims(leadTimeDict)
+        dataDic[idx] = ds
+    dataCombine = xr.combine_by_coords([dataDic[x] for x in dataDic],combine_attrs="drop_conflicts")
+    return dataCombine
+
+def read_cptdataset(leadTime, startDate, y_transform=False): #add leadTime and startDate as inputs
+    fcst_mu = combine_cptdataset(DATA_PATH,CONFIG["forecast_mu_filePattern"],y_transform=False)
+    fcst_mu = fcst_mu.sel(L=leadTime, T=startDate)
     fcst_mu_name = list(fcst_mu.data_vars)[0]
     fcst_mu = fcst_mu[fcst_mu_name]
-    fcst_var = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_var_file"])))
+    #fcst_var = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_var_file"])))
+    fcst_var = combine_cptdataset(DATA_PATH,CONFIG["forecast_var_filePattern"],dof=True,y_transform=False)
+    fcst_var = fcst_var.sel(L=leadTime, T=startDate)
     fcst_var_name = list(fcst_var.data_vars)[0]
-    fcst_var = fcst_var[fcst_var_name]
-    obs = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["obs_file"])))
+    dofVar = fcst_var["dof"]
+    fcst_var = fcst_var[fcst_var_name] #for some reason the function needed this but then the dof variable is not included
+    #obs = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["obs_file"])))
+    obs = combine_cptdataset(DATA_PATH,CONFIG["obs_filePattern"],y_transform=False)
+    obs = obs.sel(L=leadTime)
     obs_name = list(obs.data_vars)[0]
     obs = obs[obs_name]
     if y_transform:
-        hcst = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["hcst_file"])))
+        #hcst = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["hcst_file"])))
+        hcst = combine_cptdataset(DATA_PATH,CONFIG["hcst_filePattern"],y_transform=False)
+        hcst = hcst.sel(L=leadTime)
         hcst_name = list(hcst.data_vars)[0]
         hcst = hcst[hcst_name]
     else:
         hcst=None
```

Comment:
I don't understand how `y_transform` is being used here. Probably a question for @remicousin since it seems to have been this way in the previous version already. If `y_transform` is false then we don't read the hindcast data? That was probably only meant to be temporary.

Reply:
Yes, this I cannot speak on. Will have to wait for @remicousin response on that.

Reply:
is this repeat comment then saying that the config file should have y_transform set to true for all cases it is used? I am a bit confused now..

Reply:
no sorry for the confusion. Throughout the code, y_transform should point to the value in the config file. In this case at hand, that value should be true. (except at the function definition where y_transform should be set up to a default. I believed I had picked False as default, maybe that's where this False repeated everywhere came from).